core/platform/auth/firebase_auth_services_test.py

```diff
@@ -442,8 +469,20 @@ class FirebaseAccountWipeoutTests(test_utils.GenericTestBase):
     UNKNOWN_ERROR = firebase_exceptions.UnknownError('error')
 
     def setUp(self):
+        with contextlib2.ExitStack() as stack:
+            stack.enter_context(self.swap(
+                models.Registry, 'import_auth_services',
+                classmethod(lambda _: firebase_auth_services)))
+            stack.callback(FirebaseAdminSdkStub.install(self))
+
+            # Reload wipeout_service so it uses our firebase_auth_services swap.
+            python_utils.reload_module(wipeout_service)
```

Comment:
Shouldn't the swap be entered for all backend tests, not just this one?

Reply:
We don't want to switch to Firebase in this PR, so no we shouldn't. This PR is keeping the codebase in GAE-mode. As a consequence, the wipeout_service module is being loaded with GAE-services before we have a chance to swap it with Firebase.

 The only way I could get it to respect the swap is by reloading the module

Reply:
The only way I could get the module to respect the swap is by using reload. The only other solution I can think of is changing `core.platform.models` to export classes as its API rather than modules, but that would be a pretty big/sweeping change.

Reply:
I took another look and I think the main issue is that this module is depending on wipeout_service, which it really shouldn't be doing. Could the tests that are relevant to wipeout_service be moved to the wipeout_service_test file instead? That should break the dependency.

/cc @vojtechjelinek since he might also be able to provide advice on this.

Reply:
Moving the tests wouldn't fix this problem, we'd still need to run `reload_module` because wipeout_service would load with GAE-services before we can swap it out for Firebase-services.

Also I disagree that the tests belong in wipeout_service anyway, this is platform-specific behavior that needs to be respected by wipeout, and I think that belongs here.